File: frontend/src/types/projects.ts
export interface ProjectSummary {
  id: string
  title: string
  slug: string
  description: string
  difficulty: string
  project_type: string
}

export interface ProjectRubric {
  business_problem: string
  objectives: string[]
  questions: string[]
  skills: string[]
  tools: string[]
  steps: string[]
  deliverables: string[]
  evaluation_rubric: Record<string, string>
}

export interface ProjectDetail extends ProjectSummary {
  rubric: ProjectRubric
}

export interface ProjectSubmission {
  id: string
  project_id: string
  submission_url: string | null
  status: 'submitted' | 'reviewed' | 'passed'
  feedback: string | null
  submitted_at: string
  reviewed_at: string | null
}

File: backend/app/schemas/projects.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ProjectSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    title: str
    slug: str
    description: str
    difficulty: str
    project_type: str


class ProjectDetail(ProjectSummary):
    rubric: dict


class ProjectSubmissionCreate(BaseModel):
    submission_url: str = Field(min_length=1, max_length=2048)


class ProjectSubmissionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    project_id: uuid.UUID
    submission_url: str | None
    status: str
    feedback: str | None
    submitted_at: datetime
    reviewed_at: datetime | None
